source/Ship.cpp

```diff
@@ -3789,19 +3852,41 @@ const vector<weak_ptr<Ship>> &Ship::GetEscorts() const
 void Ship::AddEscort(Ship &ship)
 {
 	escorts.push_back(ship.shared_from_this());
+
+	// Check to see if the escort wants and can be carried.
+	// The escort is maybe already carried.
+	if(!ship.CanBeCarried() || ship.reservedBay)
```

Comment:
Should "ignore it" be the behavior when AddEscort is called more than once? Or should it be "last parent wins"?

I'm curious the scenarios that result in multiple ships attempting to claim the same fighter